habitat-lab/habitat/tasks/nav/nav.py

```diff
@@ -1376,6 +1417,68 @@ def _apply_velocity_action(
 
         return final_agent_state
 
+    def _apply_camera_pitch_velocity_action(
+        self,
+        camera_pitch_angular_velocity: float,
+        time_step: Optional[float] = None,
+    ):
+        """
+        Apply velocity command to the camera tilt angle for looking up and down
+        """
+        if time_step is None:
+            time_step = self._time_step
+
+        # Map velocity actions
+        self.vel_control.linear_velocity = np.array([0.0, 0.0, 0.0])
+        self.vel_control.angular_velocity = np.array(
+            [camera_pitch_angular_velocity, 0.0, 0.0]
+        )
+        sensor_states = self._sim.agents[0]._sensors  # type: ignore
+        for sensor in sensor_states:
+            sensor_state = self._sim.agents[0]._sensors[sensor].node  # type: ignore
+            break
+
+        # Construct the sensor rigid state
+        agent_mn_quat = sensor_state.rotation
+        current_rigid_state = RigidState(
+            agent_mn_quat,
+            sensor_state.translation,
+        )
+
+        # manually integrate the rigid state
+        goal_rigid_state = self.vel_control.integrate_transform(
+            time_step, current_rigid_state
+        )
+
+        # Compute the delta increase
+        delta = np.sign(camera_pitch_angular_velocity) * abs(
+            float(goal_rigid_state.rotation.angle())
+            - float(current_rigid_state.rotation.angle())
+        )
+
+        # Get the camera pitch angle
+        camera_pitch_ang = self._get_camera_pitch_angle()
+
+        # Handle the min and max pitch angles
+        if camera_pitch_ang + delta > self._camera_pitch_ang_range[1]:
+            camera_pitch_ang = self._camera_pitch_ang_range[1]
+            goal_rigid_state.rotation = mn.Quaternion.rotation(
+                mn.Rad(self._camera_pitch_ang_range[1]), mn.Vector3(1, 0, 0)
+            )
+        elif camera_pitch_ang + delta < self._camera_pitch_ang_range[0]:
+            camera_pitch_ang = self._camera_pitch_ang_range[0]
+            goal_rigid_state.rotation = mn.Quaternion.rotation(
+                mn.Rad(self._camera_pitch_ang_range[0]), mn.Vector3(1, 0, 0)
+            )
```

Comment:
Were you hoping to do this here?
```suggestion
        # Handle the min and max pitch angles
        if camera_pitch_ang + delta > self._camera_pitch_ang_range[1]:
            camera_pitch_ang = self._camera_pitch_ang_range[1]
            goal_rigid_state.rotation = mn.Quaternion.rotation(
                mn.Rad(camera_pitch_ang), mn.Vector3(1, 0, 0)
            )
        elif camera_pitch_ang + delta < self._camera_pitch_ang_range[0]:
            camera_pitch_ang = self._camera_pitch_ang_range[0]
            goal_rigid_state.rotation = mn.Quaternion.rotation(
                mn.Rad(camera_pitch_ang), mn.Vector3(1, 0, 0)
            )
```

Reply:
Good catch, thank you!